test/distributed/_shard/sharded_tensor/test_sharded_tensor.py

```diff
@@ -1097,7 +1097,11 @@ def test_state_dict(self):
         # Test save
         m._register_state_dict_hook(state_dict_hook)
         buffer = io.BytesIO()
-        torch.save(m.state_dict(), buffer)
+        mod_state_dict = state_dict = m.state_dict()
+        mod_state_keys = mod_state_dict.keys()
+        self.assertTrue("sharded_tensor1" in mod_state_keys)
+        self.assertTrue("submodule.sharded_tensor2" in mod_state_keys)
+        torch.save(mod_state_dict, buffer)
```

Comment:
Would this cause the unit test to fail without the changes below? Also, shouldn't we add test coverage for this to the load path as well since we made some changes there?